generatebundlefile/ecr_test.go

```diff
@@ -0,0 +1,240 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"testing"
+)
+
+func TestPullHelmChart(t *testing.T) {
+	tests := []struct {
+		testName        string
+		testHelmName    string
+		testHelmVersion string
+		helmLocation    string
+		wantErr         bool
+	}{
+		{
+			testName:        "Test empty name",
+			testHelmName:    "",
+			testHelmVersion: "0.1.1+9b09ef845d5d38d5201b96e32ae0be0ce2402b78",
+			helmLocation:    "",
+			wantErr:         true,
+		},
+		{
+			testName:        "Test empty version",
+			testHelmName:    "646717423341.dkr.ecr.us-west-2.amazonaws.com/hello-eks-anywhere",
+			testHelmVersion: "",
+			helmLocation:    "",
+			wantErr:         true,
+		},
+		{
+			testName:        "Test valid helm",
+			testHelmName:    "646717423341.dkr.ecr.us-west-2.amazonaws.com/hello-eks-anywhere",
+			testHelmVersion: "0.1.1+9b09ef845d5d38d5201b96e32ae0be0ce2402b78",
+			helmLocation:    fmt.Sprintf("%s/Library/Caches/helm/repository/hello-eks-anywhere-0.1.1+9b09ef845d5d38d5201b96e32ae0be0ce2402b78.tgz", os.Getenv("HOME")),
+			wantErr:         false,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.testName, func(tt *testing.T) {
+			got, err := PullHelmChart(tc.testHelmName, tc.testHelmVersion)
+			if (err != nil) != tc.wantErr {
+				t.Fatalf("PullHelmChart() error = %v, wantErr %v", err, tc.wantErr)
+			}
+			if got != tc.helmLocation {
+				t.Fatalf("PullHelmChart() = %#v\n\n\n, want %#v", got, tc.helmLocation)
+			}
+		})
+	}
+}
+
+func TestSplitECRName(t *testing.T) {
+	tests := []struct {
+		testName     string
+		testHelmName string
+		chartName    string
+		helmName     string
+		wantErr      bool
+	}{
+		{
+			testName:     "Test empty name",
+			testHelmName: "",
+			chartName:    "",
+			helmName:     "",
+			wantErr:      true,
+		},
+		{
+			testName:     "Test valid name no prefix",
+			testHelmName: "646717423341.dkr.ecr.us-west-2.amazonaws.com/hello-eks-anywhere",
+			chartName:    "hello-eks-anywhere",
+			helmName:     "hello-eks-anywhere",
+			wantErr:      false,
+		},
+		{
+			testName:     "Test valid name w/ prefix",
+			testHelmName: "646717423341.dkr.ecr.us-west-2.amazonaws.com/aws-containers/hello-eks-anywhere",
+			chartName:    "aws-containers/hello-eks-anywhere",
+			helmName:     "hello-eks-anywhere",
+			wantErr:      false,
+		},
+		{
+			testName:     "Test invalid name w/ multiple prefixes",
+			testHelmName: "646717423341.dkr.ecr.us-west-2.amazonaws.com/test/aws-containers/hello-eks-anywhere",
+			chartName:    "",
+			helmName:     "",
+			wantErr:      true,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.testName, func(tt *testing.T) {
+			got, got2, err := splitECRName(tc.testHelmName)
+			if (err != nil) != tc.wantErr {
+				t.Fatalf("splitECRName() error = %v, wantErr %v", err, tc.wantErr)
+			}
+			if got != tc.chartName || got2 != tc.helmName {
+				t.Fatalf("splitECRName() = %#v\n\n\n, want %#v %#v", got, tc.chartName, tc.helmName)
```

Comment:
```suggestion
				tt.Fatalf("splitECRName() error = %v, wantErr %v", err, tc.wantErr)
			}
			if got != tc.chartName || got2 != tc.helmName {
				tt.Fatalf("splitECRName() = %#v\n\n\n, want %#v %#v", got, tc.chartName, tc.helmName)
```
I won't comment on any more, but you get the idea.

Reply:
Awesome, fixed